src/cli/cmd.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+var path = require('path');
+var winston = require('winston');
+var async = require('async');
+var db = require('../database');
+var plugins = require('../plugins/data');
+
+function parseCommands(pluginList, callback) {
+	var commands = [];
+
+	pluginList.forEach((pluginData) => {
+		var pluginCommands = pluginData.commands || [];
+
+		pluginCommands.forEach((cmdData) => {
+			var pluginName = pluginData.id;
+			var cmdName = pluginName.replace('nodebb-plugin-', '') + ':' + cmdData.cmd;
+
+			commands.push({
+				plugin: pluginName,
+				name: cmdName,
+				description: cmdData.description,
+				options: cmdData.options,
+				action: (cmdArgs, done) => {
+					var libraryFile = cmdData.library ? cmdData.library : pluginData.library;
+					var scriptFile = path.resolve(pluginData.path, libraryFile);
+					var call = require(scriptFile)[cmdData.method];
+					call(cmdArgs, done);
+				},
+			});
+		});
+	});
+	callback(null, commands);
+}
+
+function printCommandList(commands) {
+	if (commands.length) {
+		commands.forEach((cmd) => {
+			console.log(cmd.name.green + '\t\t\t' + cmd.description);
+		});
+	} else {
+		console.warn('No commands available'.red);
+	}
+	console.log('\n');
+}
+
+function registerPluginCommands(subProgram, commands, done) {
+	commands.forEach((cmd) => {
+		var pluginCommand = subProgram
+			.command(cmd.name)
+			.description(cmd.description);
+
+		// register options
+		(cmd.options || []).forEach((opt) => {
+			pluginCommand.option(opt.flags, opt.description, null, opt.default);
+		});
+
+		// register action
+		pluginCommand.action((command, cmdArgs) => {
+			try {
+				cmd.action(cmdArgs || {}, done);
+			} catch (err) {
+				console.error('Plugin error [' + cmd.plugin + ']'.red);
```

Comment:
This is wrong. You can use template strings if you want:

    `Plugin error [${cmd.plugin}]`.red

Reply:
Oh! Sorry...
God bless pair programming! 😄 
46903e9